### PR TITLE
Pai 57 fe table

### PR DESCRIPTION
**Pull request by archanaganesan11**
Test URLs:

- Before: https://main--pricefx-eds--pricefx.hlx.live/
- After: https://pai-57-fe-table--pricefx-eds--pricefx.hlx.live/style-guide/components/table

### CODE REVIEW

**Comment by amogh-daryapurkar** [comment]
@archanaganesan11 Can you update the test URL with style guide page?

File: blocks/table/table.css [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+/* Level Table & Level Color common */
+.table .table-level, .table .levelcolor {
+  font-family: var(--ff-secondary);
+  border-collapse: collapse;
+  width: 100%;
+  height: 100%;
+}
+
+.table .table-level td,.table  .table-level th, .table .levelcolor td, .table .levelcolor th {
+  padding: 24px 32px;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:46]: Use CSS variables

File: blocks/table/table.css [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  text-align: left;
+}
+
+.table .table-level th, .table .levelcolor th {
+    font-family: var( --ff-primary);
+}
+
+.table .table-level tr:nth-child(odd), .table .levelcolor tr:nth-child(odd){
+    background-color: #E6F5FC;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:55]: Use CSS variables

File: blocks/table/table.css [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+.table .table-level th {
+  padding-top: 12px;
+  padding-bottom: 12px;
+  text-align: left;
+  background-color: #304F8C;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:62]: Use CSS variables

File: blocks/table/table.css [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  color: var(--c-white);
+}
+
+/* level color table */
+
+.table .levelcolor th {
+  padding-top: 12px;
+  padding-bottom: 12px;
+  text-align: left;
+  background-color: #3C5EA3;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:72]: Use CSS variables

File: blocks/table/table.css [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  background-color: var(--c-copy);
+  color: var(--c-white);
+}
+
+.table .table-icon td{
+  padding: 24px 32px;
+}
+
+.table .table-icon td,.table .table-icon th {
+  border: 4px solid #ddd;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:104]: Use CSS variables

File: blocks/table/table.css [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+.table .table-im-connect {
+  color: var(--c-full-black);
+  font-family: var(--ff-secondary);
+  font-size: var(--base-font-size);
+  font-style: normal;
+
+  /* font-weight: var(--fw-regular); */
+  line-height: 24px; 
+  background: #F2F2F2;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:127]: Use CSS variables

File: blocks/table/table.css [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  border-collapse: collapse;
+  max-width: 1188px;
+  overflow-x:auto;
+}
+
+/* Default Table */
+.table .table-default {
+  font-family: var(--ff-secondary);
+  border-collapse: collapse;
+  border: 2px solid black;
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.css:12]: Use CSS variables

File: styles/styles.css [hunk 8]
```diff
@@ -134,6 +134,13 @@
   --c-link-lt: #5fbbe8;
   --c-mega-menu-link: #2d9bd2;
   --c-card-subtitle: #c9559e;
+  --c-table-border: #ddd;
+  --c-table-level: #E6F5FC;
+  --c-table-level-header: #3C5EA3;
+  --c-table-background: #F2F2F2;
```
Comment (Vaitheesaran-nagalingam) [inline styles/styles.css:140]: Always use lower case for color code

File: blocks/table/table.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
+import { CIRCLEICON } from '../../scripts/constants.js';
+import { decorateButtons } from '../../scripts/aem.js';
+
+function hasNumber(myString) {
+  return /\d/.test(myString);
+}
+
+export default async function decorate(block) {
+  // eslint-disable-next-line no-unused-vars
```
Comment (Vaitheesaran-nagalingam) [inline blocks/table/table.js:9]: remove this if not required

**Pull request merged**